File: semestr2/Kurs Programowania/Lista2/WierszTrojkataPascala.cpp
#include <iostream>
#include <string.h>
#include <cstring>


using namespace std;

int* wiersz;

class WierszTrojkataPascala {
	
public:
	void correctValue(int n) {
		if (n < 0) {
			cout << n << " - niepoprawny numer wiersza" << '\n';
			exit(0);
		}
		else {
			wiersz = new int[n + 1000];
		}
	}

	void fillAnArray(int n) {
		wiersz[0] = 1;
		wiersz[n] = 1;

		for (int i = 1; i <= n; i++) {
			wiersz[i] = wiersz[i - 1] * (n - i + 1) / i;
		}
	}

	void Wspolczynnik(int m) {
		cout <<m<<" - "<<wiersz[m] <<'\n';
	}

	bool czyLiczba(string wyraz) {
		for (int i = 0; i < wyraz.length(); i++) {
			if (wyraz[i] < '0' || wyraz[i] > '9') return false;
		}
		return true;
	}

};



int main(int argc, char** argv)
{
	WierszTrojkataPascala trojkat;
	int n;

	if(trojkat.czyLiczba(argv[1]) == true) {
		n = stoi(argv[1]);
		trojkat.correctValue(n);
	}
	else {
		cout << argv[1] << " - niepoprawny numer wiersza" << "\n";
		exit(0);
	}

	trojkat.fillAnArray(n);

	for (int i = 2; i <= argc; i++)
	{
		if (trojkat.czyLiczba(argv[i]) == true) {
			int m =stoi(argv[i]);

			if (m >= 0 && m < sizeof(wiersz)+2) trojkat.Wspolczynnik(m);
			else {
				cout << m << " - liczba spoza zakresu" << "\n";
			}
		}
		else {
			cout << argv[i] << " - niepoprawny typ danych" << "\n";
		}
	}

	delete wiersz;
}

File: semestr2/Kurs Programowania/Lista2/WierszTrojkataPascala.java
public class WierszTrojkataPascala {
	
	public static int [] wiersz;

	public WierszTrojkataPascala(int n) {
		
		if(n<0) {
			System.out.println(n + " - niepoprawny numer wiersza");
			System.exit(0);
		}		
		
		wiersz = new int[n+1];
		wiersz[0] = 1;
		wiersz[n] = 1;

		for(int i = 1; i <= n; i++) {
		
			wiersz[i] = wiersz[i-1] * (n-i+1) / i;
		}
	}

	public static void Wspolczynnik(int m) {
		System.out.println(m + " - " + wiersz[m]);
	}
		


	public static void main(String[] args) {
		try {
			int n = Integer.parseInt(args[0]);
			WierszTrojkataPascala newWiersz = new WierszTrojkataPascala(n);
		}	
		catch(NumberFormatException ex) {
			System.out.println(args[0] + " - niepoprawny numer wiersza");
			System.exit(0);
		}

		for(int i = 1; i < args.length; i++) {
			try {
				int m = Integer.parseInt(args[i]);
				if(m >= 0  && m < wiersz.length) Wspolczynnik(m);
				else {
					System.out.println(m + " - liczba spoza zakresu");
			}
			}	
			catch(NumberFormatException ex) {
				System.out.println(args[i] + " - niepoprawna dana");
			}
			
		}			
		
	}
}